internal/api/http_client.go

```diff
@@ -0,0 +1,12 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/cli/go-gh/pkg/api"
+)
+
+func NewHTTPClient(opts *api.ClientOptions) *http.Client {
+	c := newHTTPClient(opts)
```

Comment:
Seems like we could just export `newHTTPClient` rather than add this function.

Reply:
No problemo, will get this done shortly.